proxy/IPAllow.h

```diff
@@ -114,8 +114,8 @@ class IpAllow : public ConfigInfo
   IpAllow(const char *config_var, const char *name, const char *action_val);
   ~IpAllow();
   void Print();
-  AclRecord *match(IpEndpoint const *ip) const;
-  AclRecord *match(sockaddr const *ip) const;
+  AclRecord *match(IpEndpoint const *ip, bool is_dest_ip) const;
```

Comment:
Let's change this to an `enum` like

```
enum class match_t { SRC_ADDR, DST_ADDR };
```

This should make the call site clearer.